src/toGamut.js

```diff
@@ -179,21 +179,24 @@ export function toGamutCSS (origin, { space = origin.space }) {
 	}
 	let min = 0;
 	let max = origin_OKLCH.coords[1];
-
 	let min_inGamut = true;
-	let current;
+	let current = clone(origin_OKLCH);
+    let clipped = clip(current);
+
+    let E = deltaEOK(clipped, current);
+    if (E < JND) {
+		return clipped;
+	}
 
 	while ((max - min) > ε) {
 		const chroma = (min + max) / 2;
-		current = clone(origin_OKLCH);
 		current.coords[1] = chroma;
 		if (min_inGamut && inGamut(current, space)) {
 			min = chroma;
-			continue;
 		}
 		else if (!inGamut(current, space)) {
```

Comment:
I notice that both the codepen and ColorAide are using a plain `else` here and not including the `if (!inGamut(current, space))` -- is that correct (and the spec should be adjusted), or do we want to leave this here?

Reply:
I understand the confusion as if you are reading the spec is says:

> otherwise, if inGamut(current) is false carry out these steps:

It should be noted that this is not a false statement, you just don't need to explicit check it because the state is implied.

Once the lower limit is out of gamut (`min_inGamut` is `false`), you don't need to check if you are in gamut anymore as it is implied you are out of gamut. If `min_inGamut` is `true`, then you have to do the "in gamut" check, and then if it yields `false` you know you are not in gamut.  In short `if (min_inGamut && inGamut(current, space))` is fails, it is implied that you are out of gamut, so all you need is an `else`.

Reply:
@facelessuser Thanks -- that makes sense. I removed the unnecessary check.